beamformer/beamform_coeffs/beamformcoeff_kernel.py

```diff
@@ -11,161 +11,151 @@
+        # Compute indexes for output matrix
+        ibatchindex = ithreadindex_x // (pols * n_channels * n_beams * n_ants)
+        ibatchindex_rem = ithreadindex_x % (pols * n_channels * n_beams * n_ants)
 
-    ipolindex = ibatchindex_rem // (n_channels * n_beams * n_ants)
-    ipolindex_rem = ibatchindex_rem % (n_channels * n_beams * n_ants)
+        ipolindex = ibatchindex_rem // (n_channels * n_beams * n_ants)
+        ipolindex_rem = ibatchindex_rem % (n_channels * n_beams * n_ants)
 
-    ichannelindex = ipolindex_rem // (n_beams * n_ants)
-    ichannelindex_rem = ipolindex_rem % (n_beams * n_ants)
+        ichannelindex = ipolindex_rem // (n_beams * n_ants)
+        ichannelindex_rem = ipolindex_rem % (n_beams * n_ants)
 
-    ibeamindex = ichannelindex_rem // (n_ants)
-    ibeamindex_rem = ichannelindex_rem % (n_ants)
+        ibeamindex = ichannelindex_rem // (n_ants)
+        ibeamindex_rem = ichannelindex_rem % (n_ants)
 
-    ibeam_matrix = ibeamindex * 2
-    iant_matrix = ibeamindex_rem * 2
+        ibeam_matrix = ibeamindex * 2
+        iant_matrix = ibeamindex_rem * 2
 
-    # Store steering coefficients in output matrix
-    coeffs[ibatchindex][ipolindex][ichannelindex][iant_matrix][ibeam_matrix] = steering_coeff_correct_real
-    coeffs[ibatchindex][ipolindex][ichannelindex][iant_matrix][ibeam_matrix + 1] = steering_coeff_correct_imag
+        # Store steering coefficients in output matrix
+        coeffs[ibatchindex][ipolindex][ichannelindex][iant_matrix][ibeam_matrix] = steering_coeff_correct_real
+        coeffs[ibatchindex][ipolindex][ichannelindex][iant_matrix][ibeam_matrix + 1] = steering_coeff_correct_imag
 
-    coeffs[ibatchindex][ipolindex][ichannelindex][iant_matrix + 1][ibeam_matrix] = -steering_coeff_correct_imag
-    coeffs[ibatchindex][ipolindex][ichannelindex][iant_matrix + 1][ibeam_matrix + 1] = steering_coeff_correct_real
+        coeffs[ibatchindex][ipolindex][ichannelindex][iant_matrix + 1][ibeam_matrix] = -steering_coeff_correct_imag
+        coeffs[ibatchindex][ipolindex][ichannelindex][iant_matrix + 1][ibeam_matrix + 1] = steering_coeff_correct_real
 
 
 class BeamformCoeffKernel:
-    """Class for beamform complex multiplication."""
-
-    # def __init__(self, batches, n_channels, n_blocks, samples_per_block, n_ants, xeng_id):
+    """Class for beamform complex multiplication.
+
+    Parameters
+    ----------
+    delay_vals:
+        Data matrix of delay values.
+    batches:
+        Number of batches to process.
+    pols:
+        Number of polarisations.
+    num_channels:
+        The number of channels the XEng core will process.
+    total_channels:
+        The total number of channels in the system.
+    n_blocks:
+        Number of blocks into which samples are divided in groups of 16
+    samples_per_block:
+        Number of samples to process per sample-block
+    n_ants:
+        The number of antennas from which data will be received.
+    num_beams:
```

Comment:
Name doesn't match the parameter
```suggestion
    n_beams:
```